tests/unit/test_routes.py

```diff
@@ -371,7 +373,122 @@ def test_update_resource(module_client, module_db, fake_auth_from_oc):
     assert (response.status_code == 404)
 
 
-def test_bad_requests(module_client, module_db, fake_auth_from_oc):
+def test_search(module_client, module_db, fake_auth_from_oc, fake_algolia_save, fake_algolia_search):
+    client = module_client
+
+    first_term = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    apikey = get_api_key(client)
+
+    # Create resource and find it in the search results.
+    resource = client.post("/api/v1/resources",
+                          json=dict(
+                              name=f"{first_term}",
+                              category="Website",
+                              url=f"{first_term}",
+                              paid=False,
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    result = client.get(f"/api/v1/search?q={first_term}")
+
+    assert (resource.status_code == 200)
+    assert (result.status_code == 200)
+    assert (result.json['data'][0]['url'] == resource.json['data'].get('url'))
+
+    # Update the resource and test that search results reflect changes
+    updated_term = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
```

Comment:
Since we're using this more than once, why not make a utility function called `random_string()`? That would be a bit more readable because you don't have to try to grok how it's creating the random string to know that that's what it's doing.